libr/core/cmd_print.c

```diff
@@ -5659,6 +5659,74 @@ l = use_blocksize;
 				}
 			}
 			break;
+		case 't': // "pst"
+			if (l > 0) {
+				ut8 *var = malloc (len);
+				size_t len_charset = sdb_count (core->anal->charset_db);
+				r_charset_encode_str (var, core->block, len, core->custo.custom_charset, len_charset);
+				r_print_string (core->print, core->offset, var, len, R_PRINT_STRING_ZEROEND);
+				free(var);
+				}
+			break;
+		case 'e': // "pse"
+			if (input[2] == ' ') {
+				const char *fname = r_str_trim_head_ro (input + 3);
+				char *tmp = r_str_newf (R_JOIN_2_PATHS (R_JOIN_2_PATHS (R2_HOME_DATADIR, "util"), "%s"), fname);//todo: add R2_HOME_UTIL
+				char *home = r_str_home (tmp);
+				//free (tmp);
+				tmp = r_str_newf (R_JOIN_2_PATHS (R_JOIN_2_PATHS (R2_SDB, "util"), "%s"), r_str_newf (R_JOIN_2_PATHS ("d", "%s"), fname));
+				char *path = r_str_r2_prefix (tmp);
+				if (!sdb_query_file (core->anal->charset_db, path)) {
+					eprintf ("Path: %s not found\n", path);
+				}
+				free (home);
+				free (path);
+				free (tmp);
```

Comment:
should this return here or continue? seems kinda spooky

Reply:
@crowell I do not see any return at the end of any case in this function. Then I do not think so. Your question makes me realize I should put the charset processing in the if condition. Thank you very much.